### DISCUSSION TITLE
control the mapping property names without notation ?

### DISCUSSION THREAD

**Discussion by Nir-An** [spatie/laravel-data#172] [Q&A] [answered]
If I want to validate the properties I can use notations or function [rules](https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/request-to-data-object#content-using-validation) without notation at all.
> It is also possible to write rules down in a dedicated method on the data object. This can come in handy when you want to construct a custom rule object which isn't possible with attributes:

Like the rules function, there is a way to control the[ mapping property names](https://spatie.be/docs/laravel-data/v2/as-a-resource/from-data-to-resource#content-mapping-property-names) without notation? 

something like function or flag?

Thanks for a beautiful project.

**Comment by rubenvanassche** [comment, marked as answer]
So you want to map the names of properties within a dedicated function?

What would be the use case for this?

If we want to implement such a thing, the best way would be to overwrite the `$inputMappedName` and `$outputMappedName` in `DataProperty`. The problem then would be that a cached version of such will probably not contain the correct value for these properties.

**Reply by Nir-An** [reply]
> What would be the use case for this?
 
Some developers don't want to use "active" annotations for changing properties.
And maybe it makes a performance issue because of the Reflection payload,
but I'm not sure about the second one

Nevertheless, it is so close to java spring :smile:  
and it makes it a little bit hard to understand and debug for beginners, 
but I understand your point.

very useful package anyway! thanks!
